src/sctools/test/test_fastq.py

```diff
@@ -188,3 +188,57 @@ def test_fields_populate_properly(reader_all_compressions):
         assert all(i in alphabet for i in record.sequence.strip())
         assert record.name2 == name2_string
         assert all(i in ascii_chars for i in record.quality.strip())
+
+
+# TEST BarcodeGeneratorWithCorrectedCellbarcodes
+
+@pytest.fixture(scope='function')
+def embedded_barcode_generator():
+    cell_barcode = fastq.EmbeddedBarcode(start=0, end=16, quality_tag='CY', sequence_tag='CR')
+    molecule_barcode = fastq.EmbeddedBarcode(start=16, end=26, quality_tag='UY', sequence_tag='UR')
+    return fastq.EmbeddedBarcodeGenerator(data_dir + 'test_r1.fastq.gz',
+                                          [cell_barcode, molecule_barcode])
+
+
+@pytest.fixture(scope='function')
+def barcode_generator_with_corrected_cell_barcodes():
+    cell_barcode = fastq.EmbeddedBarcode(start=0, end=16, quality_tag='CY', sequence_tag='CR')
+    molecule_barcode = fastq.EmbeddedBarcode(start=16, end=26, quality_tag='UY', sequence_tag='UR')
+    return fastq.BarcodeGeneratorWithCorrectedCellBarcodes(
+        data_dir + 'test_r1.fastq.gz', cell_barcode, data_dir + '1k-august-2016.txt',
+        molecule_barcode)
+
+
+def test_embedded_barcode_generator_produces_outputs_of_expected_size(embedded_barcode_generator):
+    for cell_seq, cell_qual, umi_seq, umi_qual in embedded_barcode_generator:
+
+        # test cell tags
+        assert cell_seq[0] == 'CR'
+        assert len(cell_seq[1]) == 16
```

Comment:
Can you replace these hard-coded numbers with variables? You can make them properties of the test class

Reply:
Added some comments, but generally I want to avoid unnecessary verbosity by duplicating all of these.